src/Parser.js

```diff
@@ -844,6 +844,10 @@ export default class Parser {
             if (mode) {
                 this.switchMode(mode);
             }
+            // Start a new namespace
+            if (!optional) {
+                this.gullet.pushNamespace();
```

Comment:
It seems they are allowed in optional groups and their scope is limited to the group, e.g. `\sqrt[\def\foo{3}\foo]{x}` works but `\sqrt[\def\foo{3}]{x\foo}\foo` doesn't.

Reply:
Doesn't a optional group start new namespace?

Reply:
Sorry, I forgot to fix this according to @ylemkimon's experiments. (I had assumed optional groups didn't make their own groups because they didn't use `{}` notation.) I'll remove the `if (!optional)`.